src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,204 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.value = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            linkFirst(value);
+            return;
+        }
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        }
+        if (index == size) {
+            linkLast(value);
+            return;
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] listToAdd = list.toArray();
+        int countToAdd = listToAdd.length;
+
+        for (int i = 0; i < countToAdd; i++) {
+            if (first == null) {
+                linkFirst((T) listToAdd[i]);
+            } else {
+                linkLast((T) listToAdd[i]);
+            }
+        }
```

Comment:
can we use our `add(T value)` method here?

Reply:
sure!